catalog/ui/search-ui/simple/src/test/java/org/codice/ddf/ui/searchui/simple/properties/TestUiConfigurationPropertiesFactory.java

```diff
@@ -0,0 +1,95 @@
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TestUiConfigurationPropertiesFactory {
+
+  @Mock private ConfigurationAdmin configurationAdmin;
+
+  private Optional<BrandingRegistry> branding = Optional.empty();
+
+  @Test
+  public void testConfigurationDoesNotExist() throws IOException, InvalidSyntaxException {
+    when(configurationAdmin.listConfigurations("(service.pid=ddf.platform.ui.config)"))
+        .thenReturn(null);
+    UiConfigurationPropertiesFactory.getInstance().setConfigurationAdmin(configurationAdmin);
+    UiConfigurationProperties props =
+        UiConfigurationPropertiesFactory.getInstance().getProperties();
+    assertAllBlankProps(props);
+  }
+
+  @Test
+  public void testExceptionIsHandled() throws IOException, InvalidSyntaxException {
+    when(configurationAdmin.listConfigurations("(service.pid=ddf.platform.ui.config)"))
+        .thenThrow(new IOException());
+    UiConfigurationPropertiesFactory.getInstance().setConfigurationAdmin(configurationAdmin);
+    UiConfigurationProperties props =
+        UiConfigurationPropertiesFactory.getInstance().getProperties();
+    assertAllBlankProps(props);
+  }
+
+  @Test
+  public void testPropsCreatedProperly() throws IOException, InvalidSyntaxException {
+    DictionaryMap<String, Object> configProps = new DictionaryMap<>();
+    configProps.put("header", "Test Header");
+    configProps.put("footer", "Test Footer");
+    configProps.put("systemUsageEnabled", true);
+
+    Configuration configuration = mock(Configuration.class);
+    Configuration[] configurations = {configuration};
+    when(configurationAdmin.listConfigurations("(service.pid=ddf.platform.ui.config)"))
+        .thenReturn(configurations);
+    when(configurationAdmin.getConfiguration("ddf.platform.ui.config", null))
+        .thenReturn(configuration);
+    when(configuration.getProperties()).thenReturn(configProps);
+
+    UiConfigurationPropertiesFactory.getInstance().setConfigurationAdmin(configurationAdmin);
+    UiConfigurationProperties props =
+        UiConfigurationPropertiesFactory.getInstance().getProperties();
+
+    assertThat(props.getHeader(), is("Test Header"));
+    assertThat(props.getFooter(), is("Test Footer"));
+    assertThat(props.getSystemUsageEnabled(), is(true));
+  }
+
+  private void assertAllBlankProps(UiConfigurationProperties props) {
+    assertThat(props.getProductName(), is(""));
+    assertThat(props.getHeader(), is(""));
+    assertThat(props.getFooter(), is(""));
+    assertThat(props.getColor(), is(""));
+    assertThat(props.getBackground(), is(""));
+    assertThat(props.getSystemUsageTitle(), is(""));
+    assertThat(props.getSystemUsageMessage(), is(""));
+    assertThat(props.getSystemUsageEnabled(), is(false));
+    assertThat(props.getSystemUsageOncePerSession(), is(false));
+  }
+}
```

Comment:
❗️ We also need tests for the `getProductName()` method, one for when it is set and one for when it isn't.
❗️ We should also make sure that properties that we properly handle properties that have the wrong type, e.g., header isn't a `String` and system usage enabled isn't a `Boolean`.

Reply:
Fair! I will add these.